src/Microsoft.ML.Core/Data/DataKind.cs

```diff
@@ -205,7 +205,7 @@ public static bool TryGetDataKind(this Type type, out DataKind kind)
                 kind = DataKind.R4;
             else if (type == typeof(Double))
                 kind = DataKind.R8;
-            else if (type == typeof(DvText))
+            else if (type == typeof(DvText) || type == typeof(string))
```

Comment:
This is inappropriate. Actually the below `bool` below is wrong as well.

Reply:
There's another utility function I mentioned in the other code review, `GetDataType`, in the `ModuleArgs.cs` that has this more "informal" mapping. But assuredly the core definitions of `IDataView` and its supporting type structures are not the place to have this sort of stuff. :D

---
In reply to: [188000280](https://github.com/dotnet/machinelearning/pull/142#discussion_r188000280) [](ancestors = 188000280)